File: src/GymManagement.Domain/Entities/UserSchedule.cs
namespace GymManagement.Domain.Entities
{
    public class UserSchedule : BaseEntity
    {
        public int UserId { get; set; }
        public int? TrainerId { get; set; }
        public int WorkoutPlanId { get; set; }
        public ScheduleType ScheduleType { get; set; }
        public DayOfWeek DayOfWeek { get; set; }
        public TimeSpan StartTime { get; set; }
        public TimeSpan EndTime { get; set; }
        public bool IsActive { get; set; } = true;

        // Navigation properties
        public User User { get; set; } = null!;
        public Trainer? Trainer { get; set; }
        public WorkoutPlan WorkoutPlan { get; set; } = null!;
    }
}

File: gym_client/src/types/userSchedule.ts
export type ScheduleType =
  | 'OneMusclePerDay'
  | 'TwoMusclesPerDay'
  | 'ThreeMusclesPerDay'
  | 'FullBody'
  | 'Custom'

export interface UserScheduleDto {
  id: number
  userId: number
  userName: string
  trainerId?: number | null
  trainerName?: string | null
  workoutPlanId: number
  workoutPlanName: string
  scheduleType: ScheduleType
  dayOfWeek: number
  startTime: string
  endTime: string
  isActive: boolean
  assignedAt: string
  lastUpdatedAt?: string | null
}

export interface CreateUserScheduleDto {
  userId: number
  trainerId?: number | null
  workoutPlanId: number
  scheduleType: ScheduleType
  dayOfWeek: number
  startTime: string
  endTime: string
}

export interface AssignWorkoutPlanDto {
  userId: number
  trainerId?: number | null
  workoutPlanId: number
  scheduleType: ScheduleType
  dayOfWeek: number
  startTime: string
  endTime: string
  deactivateExistingAssignments?: boolean
}
